### PR TITLE
utils: don't panic when no row returned for SHOW MASTER STATUS

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1732

### What is changed and how it works?
give an error with message, to maintain "When the returned error is nil, the gtid.Set must be not nil"

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: pkg/utils/db.go [hunk 1]
```diff
@@ -201,26 +202,32 @@ func GetMasterStatus(ctx context.Context, db *sql.DB, flavor string) (gmysql.Pos
...
+	}
 
-		binlogPos = gmysql.Position{
-			Name: binlogName,
-			Pos:  pos,
-		}
+	if len(rowColumns) == 5 {
+		err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr, &gtidStr)
+	} else {
+		err = rows.Scan(&binlogName, &pos, &nullPtr, &nullPtr)
```
Comment (lichunzhu) [inline pkg/utils/db.go:212]: Do we need `gtidStr` for mariaDB?
https://github.com/pingcap/dumpling/blob/master/v4/export/metadata.go#L137

File: pkg/utils/db.go [hunk 2]
```diff
@@ -201,26 +202,32 @@ func GetMasterStatus(ctx context.Context, db *sql.DB, flavor string) (gmysql.Pos
...
+
+	gs, err = gtid.ParserGTID(flavor, gtidStr)
+	if err != nil {
+		return binlogPos, gs, err
+	}
+	if rows.Next() {
+		log.L().Warn("SHOW MASTER STATUS returns more than one row, will only use first row")
+	}
+
 	if rows.Err() != nil {
```
Comment (lichunzhu) [inline pkg/utils/db.go:231]: We'd better close `rows` before checking `rows.Err()`.

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: pkg/utils/db.go [hunk 3]
```diff
@@ -201,26 +202,32 @@ func GetMasterStatus(ctx context.Context, db *sql.DB, flavor string) (gmysql.Pos
 		pos        uint32
```
Comment (Ehco1996) [inline pkg/utils/db.go:202]: can we add `show master status` result  from mariadb in comment ?

something like
```
		For MariaDB,SHOW MASTER STATUS:
		+--------------------+----------+--------------+------------------+
		| File               | Position | Binlog_Do_DB | Binlog_Ignore_DB |
		+--------------------+----------+--------------+------------------+
		| mariadb-bin.000016 |      475 |              |                  |
		+--------------------+----------+--------------+------------------+
		SELECT @@global.gtid_binlog_pos;
		+--------------------------+
		| @@global.gtid_binlog_pos |
		+--------------------------+
		| 0-1-2                    |
		+--------------------------+
```

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by Ehco1996** [comment]
LGTM

**Pull request merged**